packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -611,32 +322,30 @@ export abstract class QueryEditorStore {
         },
       },
     );
+    this.queryCreatorState = new QueryCreatorState(this, undefined);
   }
 
-  get isViewProjectActionDisabled(): boolean {
-    return false;
-  }
+  abstract get label(): string;
 
-  get isSaveActionDisabled(): boolean {
+  get isViewProjectActionDisabled(): boolean {
     return false;
   }
 
-  setSaveAsState(val: QuerySaveAsState | undefined): void {
-    this.saveAsState = val;
-  }
-
-  setSaveState(val: QuerySaveState | undefined): void {
-    this.saveState = val;
-  }
-
-  setRenameState(val: QueryRenameState | undefined): void {
-    this.renameState = val;
+  get queryBuilderState(): QueryBuilderState {
```

Comment:
this change and the change to `_queryBuilderState` actually make it much harder to follow the code, I don't think `QueryEditorStore.queryBuilderState` gets used in too many places, can we just leave it like before? Thanks!